### DISCUSSION TITLE
Issues building a working VGA uf2

### DISCUSSION THREAD

**Discussion by ixmeins** [picocomputer/community#15] [Q&A]
Is anyone successfully building the VGA uf2 file? I have the dev environment all setup and I can run an error free make of the rp6502 project which does generate uf2 files. The RIA uf2 works fine but the VGA uf2 appears too small by about 40k (compared to public releases) and it does not work when installed. It just produces half a screen of blue. Any hints appreciated.

**Comment by rumbledethumps** [comment]
The half blue screen is part of the pico-extras video library. It means the raster data wasn't generated in time. There's a hot loop in the RIA that needs an override to "-O2", which neither debug or release builds use, so maybe the VGA needs a similar hack somewhere.

**Reply by ixmeins** [reply]
Okay, so how do you build it?

**Reply by rumbledethumps** [reply]
Just the usual F7 here. Perhaps I have a different version of GCC. It'll probably start happening for me too next time I get into the VGA code (soon). Once I corner it, I can fix it.

arm-none-eabi-gcc (15:10.3-2021.07-4) 10.3.1 20210621 (release)

**Reply by ixmeins** [reply]
Ok I'll go check versions etc. (edit: that's the same version of GCC) Sorry I should clarify the issue as it works in text mode but not once it enters graphics mode I should have reread my post earlier. I'll go play when I get a chance.

**Comment by rumbledethumps** [comment]
Needed to tell optimizer about some volatiles:
https://github.com/picocomputer/rp6502/commit/44a13dae4482bdab255afffc1f95c794561464b3

**Reply by ixmeins** [reply]
180 line mode and pixel 0 now works but 240 line mode reverts to the colour bars (no more half blue screen).
